src/cdk/datetime/dateadapter/date-adapter.ts

```diff
@@ -10,23 +10,23 @@ import {inject, InjectionToken, LOCALE_ID} from '@angular/core';
 import {Observable, Subject} from 'rxjs';
 
 /** InjectionToken for datepicker that can be used to override default locale code. */
-export const MAT_DATE_LOCALE = new InjectionToken<string>('MAT_DATE_LOCALE', {
```

Comment:
How about the users who are using the `MAT_DATE_LOCALE`?

Reply:
I changed all of the dependencies on `MAT_DATE_LOCALE` to `CDK_DATE_LOCALE`

Reply:
But `MAT_DATE_LOCALE` is a public API and may be used by external users.

Reply:
Would it be better to leave this as `MAT_DATE_LOCALE` in the DateAdapter CDK then?

Reply:
In order to make this backward compatible, we have to re-export these from `src/lib/core/datetime/public-api.ts`:

```ts
export {
  DateAdapter, 
  CDK_DATE_LOCALE as MAT_DATE_LOCALE, 
  CDK_DATE_LOCALE_FACTORY as MAT_DATE_LOCALE_FACTORY
} from '@angular/cdk/datetime'
```